### PR TITLE
Improve Prow development docs a bit.

### PR DESCRIPTION
**Pull request by cjwagner**
/assign @stevekuznetsov @fejta

### CODE REVIEW

File: prow/build_test_update.md [hunk 1]
```diff
@@ -20,7 +20,41 @@ bazel build //prow/cmd/hook
...
+## How to update the cluster
+
+Any modifications to Go code will require redeploying the affected binaries.
+Assuming your prow components have multiple replicas, this will result in no downtime.
+
+Update your deployment (optionally build/pushing the image) to a new image with:
+```shell
+# export PROW_REPO_OVERRIDE=gcr.io/k8s-prow  # optionally override project
+push.sh  # Build and push the current repo state.
+bump.sh --list  # Choose a recent published version
```
Comment (cjwagner) [inline prow/build_test_update.md]: This is the only part of the "How to update the cluster" section that I actually updated. The rest I just moved.

File: prow/build_test_update.md [hunk 2]
```diff
@@ -20,7 +20,41 @@ bazel build //prow/cmd/hook
 bazel test //prow/plugins/lgtm:go_default_test
 ```
 
-**TODO**(cjwagner): Unify and document how to run prow components locally.
+### How to test a plugin
+
+If you are making changes to a Prow plugin you can test the new behavior by sending fake webhook events with [`phony`](/prow/cmd/phony#phony).
```
Comment (stevekuznetsov) [inline prow/build_test_update.md]: Maybe 

```
sending fake webhooks to `hook` with `phony`
```

?

File: prow/build_test_update.md (hunk 1 above)
Comment (stevekuznetsov) [reply prow/build_test_update.md]: Is this using @Katharine 's new tool?

File: prow/build_test_update.md [hunk 3]
```diff
@@ -20,7 +20,41 @@ bazel build //prow/cmd/hook
...
+export BUILD_CLUSTER_OVERRIDE=gke_my-project_my-zone_my-cluster # or whatever the correct value is
+
+# Generally just do
+bazel run //prow/cluster:production.apply # deploy everything
+
+# In case of an emergency hook update
+bazel run //prow/cluster:hook.apply # just update hook
+
+# This is equivalent to doing the following with kubectl directly:
+kubectl config use-context gke_my-project_my-zone_my-cluster
```
Comment (stevekuznetsov) [inline prow/build_test_update.md]: Could we de-GKE-ify this?

File: prow/jobs.md [hunk 4]
```diff
@@ -118,9 +118,8 @@ to Prow in some other context, like `/close`. It is similarly not suggested to d
 
 #### Posting GitHub Status Contexts
 
-Jobs that run will always post a status context to the commit under test in GitHub.
-Jobs that run conditionally but do not match the content of the pull request will
-_not_ post "Skipped" status contexts to the pull request.
+Presubmits that do not have `skip_report: true` will always post a status context to the commit under test in GitHub.
```
Comment (stevekuznetsov) [inline prow/jobs.md]: nit: text wrapping length

Comment (stevekuznetsov) [reply prow/jobs.md]: Technically now postsubmits can publish

Comment (stevekuznetsov) [reply prow/jobs.md]: Could we flip the negative here to have the common case first? 

```
Jobs will post a status context in GitHub to the commit under test, unless the job is configured with `skip_report: true`
```

File: prow/build_test_update.md (hunk 1 above)
Comment (cjwagner) [reply prow/build_test_update.md]: No, we just moved the `./bump.sh --push` functionality into the `./push.sh` script AFAICT.

File: prow/build_test_update.md (hunk 3 above)
Comment (cjwagner) [reply prow/build_test_update.md]: Sure. What do you suggest?

Comment (cjwagner) [reply prow/build_test_update.md]: I switched to `my-k8s-cluster-context`

File: prow/build_test_update.md (hunk 2 above)
Comment (cjwagner) [reply prow/build_test_update.md]: Done.

File: prow/jobs.md (hunk 4 above)
Comment (cjwagner) [reply prow/jobs.md]: Done.

The text wrapping still displays correctly if we don't manually manage the line length. I'm not a fan of shuffling paragraph text around when I add a few words to the first sentence so whenever I touch markdown text that manually manages line length I switch it to long lines and let GH manage the wrapping.

**Pull request merged**